src/plugins/maven-workspace.ts

```diff
@@ -57,72 +66,26 @@ const XPATH_PROJECT_VERSION =
   '/*[local-name()="project"]/*[local-name()="version"]';
 const XPATH_PROJECT_DEPENDENCIES =
   '/*[local-name()="project"]/*[local-name()="dependencies"]/*[local-name()="dependency"]';
+const XPATH_PROJECT_DEPENDENCY_MANAGEMENT_DEPENDENCIES =
+  '/*[local-name()="project"]/*[local-name()="dependencyManagement"]/*[local-name()="dependencies"]/*[local-name()="dependency"]';
 
 export class MavenWorkspace extends WorkspacePlugin<MavenArtifact> {
+  private considerAllArtifacts: boolean;
+  constructor(
+    github: GitHub,
+    targetBranch: string,
+    repositoryConfig: RepositoryConfig,
+    options: MavenWorkspacePluginOptions = {}
+  ) {
+    super(github, targetBranch, repositoryConfig, options);
+    this.considerAllArtifacts = options.considerAllArtifacts ?? false;
+  }
   private async fetchPom(path: string): Promise<MavenArtifact | undefined> {
     const content = await this.github.getFileContentsOnBranch(
       path,
       this.targetBranch
     );
-    const document = new dom.DOMParser().parseFromString(content.parsedContent);
-
-    const groupNodes = xpath.select(XPATH_PROJECT_GROUP, document) as Node[];
-    if (groupNodes.length === 0) {
-      this.logger.warn(`Missing project.groupId in ${path}`);
-      return;
-    }
-    const artifactNodes = xpath.select(
-      XPATH_PROJECT_ARTIFACT,
-      document
-    ) as Node[];
-    if (artifactNodes.length === 0) {
-      this.logger.warn(`Missing project.artifactId in ${path}`);
-      return;
-    }
-    const versionNodes = xpath.select(
-      XPATH_PROJECT_VERSION,
-      document
-    ) as Node[];
-    if (versionNodes.length === 0) {
-      this.logger.warn(`Missing project.version in ${path}`);
-      return;
-    }
-    const dependencies: Gav[] = [];
-    const testDependencies: Gav[] = [];
-    for (const dependencyNode of xpath.select(
-      XPATH_PROJECT_DEPENDENCIES,
-      document
-    ) as Node[]) {
-      const parsedNode = parseDependencyNode(dependencyNode);
-      if (!parsedNode.version) {
-        continue;
-      }
-      if (parsedNode.scope === 'test') {
-        testDependencies.push({
-          groupId: parsedNode.groupId,
-          artifactId: parsedNode.artifactId,
-          version: parsedNode.version,
-        });
-      } else {
-        dependencies.push({
-          groupId: parsedNode.groupId,
-          artifactId: parsedNode.artifactId,
-          version: parsedNode.version,
-        });
-      }
-    }
-    const groupId = groupNodes[0].firstChild!.textContent!;
-    const artifactId = artifactNodes[0].firstChild!.textContent!;
-    return {
-      path,
-      groupId,
-      artifactId,
-      name: `${groupId}:${artifactId}`,
-      version: versionNodes[0].firstChild!.textContent!,
-      dependencies,
-      testDependencies,
-      pomContent: content.parsedContent,
-    };
+    return parseMavenArtifact(content.parsedContent, path, this.logger);
   }
 
   protected async buildAllPackages(
```

Comment:
I don't find that the method name `buildAllPackages` describes well what's happening, which if I understand correctly is we're essentially building a graph of dependencies that need to be updated.

It would be helpful for me if the graph building logic was pulled into a helper, called something like `buildUpdateGraph`, and if this method had a detailed comment that explained the behavior. For future generations working in the codebase, with some of these more complex algorithms, I think it would even be useful to provide an ascii graph (with something like https://asciiflow.com) which shows how the dependency graph is built.

Reply:
This is an abstract method declared by the base `workspace` plugin. The idea here is to scan the entire repository for packages that could interact with each other. In this case, the template type `T` is a `MavenArtifact`. The method is implemented by the per-language workspace plugins and is documented [here](https://github.com/googleapis/release-please/blob/main/src/plugins/workspace.ts#L224-L229).

This method does not build a graph - shared code in the base `workspace` plugin is building the graph of all the artifacts detected from this method.

Reply:
Adding a line 

```
"considerAllArtifacts": true
```

is fine to me.

Reply:
(Not a big deal) Are you sure it's good idea to have "considerAllArtifacts" and "type" at the same level? I felt natural to have "configuration" field to hold an object like below:

```
  "plugins": [
    {
      "type": "maven-workspace",
      "configuration" : {
        "considerAllArtifacts": true
      }
    }
  ]
```

Reply:
> However, I also feel that I don't think of a situation where I want to set the parameter to `false`

After further consideration, I could see us defaulting this parameter to true. It could cause expected pom.xml files to be touched if they exist within a directory that is managed by release-please, but in those cases, it could be configured to only consider the root pom.xml for each configured component.

> I felt natural to have "configuration" field to hold an object like below

This is probably a better design, but all the other configurable plugins have their options at the same level as `type`. For consistency, we should probably keep them at the same level.

Reply:
Thank you for explanation.

Reply:
@suztomo Switched this option to default to `true`. You can now opt-out of this behavior by setting the `considerAllArtifacts` to `false`.